tools/integrations/schema_generation.py

```diff
@@ -80,51 +89,212 @@ class _FluidProperties(Flow360BaseModel):
         options=["From density and temperature", "From pressure and temperature"],
     )
 
+    class SchemaConfig(Flow360BaseModel.SchemaConfig):
+        field_properties = {
+            "temperature": ("field", "unitInput"),
+            "density": ("field", "unitInput"),
+            "pressure": ("field", "unitInput"),
+        }
+        root_property = "properties/fluidProperties/anyOf"
+
 
 class _InitialConditions(Flow360BaseModel):
     initial_conditions: Union[FreestreamInitialCondition, ExpressionInitialCondition] = pd.Field(
         alias="initialConditions", options=["Freestream", "Expression"]
     )
 
+    class SchemaConfig(Flow360BaseModel.SchemaConfig):
+        root_property = "properties/initialConditions/anyOf"
 
-write_schemas(fl.NavierStokesSolver, "navier-stokes")
-write_schemas(fl.Geometry, "geometry")
-write_schemas(fl.SlidingInterface, "sliding-interface")
-write_schemas(fl.TransitionModelSolver, "transition-model")
-write_schemas(fl.HeatEquationSolver, "heat-equation")
-write_schemas(fl.PorousMedium, "porous-media")
-write_schemas(fl.ActuatorDisk, "actuator-disk")
-write_schemas(fl.BETDisk, "bet-disk")
-write_schemas(fl.VolumeOutput, "volume-output")
-write_schemas(fl.AeroacousticOutput, "aeroacoustic-output")
-write_schemas(fl.SliceOutput, "slice-output", swap_fields={"slices": fl.Slices.flow360_schema()})
-write_schemas(
-    fl.MonitorOutput, "monitor-output", swap_fields={"monitors": fl.Monitors.flow360_schema()}
-)
-write_schemas(
-    fl.SurfaceOutput, "surface-output", swap_fields={"surfaces": fl.Surfaces.flow360_schema()}
-)
-write_schemas(
-    fl.IsoSurfaceOutput,
-    "iso-surface-output",
-    swap_fields={"isoSurfaces": fl.IsoSurfaces.flow360_schema()},
-)
 
-write_schemas(_Freestreams, "freestream", root_property=["properties", "freestream", "anyOf"])
-write_schemas(
-    _TimeSteppings, "time-stepping", root_property=["properties", "timeStepping", "anyOf"]
-)
-write_schemas(
-    _FluidProperties, "fluid-properties", root_property=["properties", "fluidProperties", "anyOf"]
-)
-write_schemas(
-    _TurbulenceModelSolvers, "turbulence-model", root_property=["properties", "solver", "anyOf"]
-)
-write_schemas(
-    _InitialConditions,
-    "initial-conditions",
-    root_property=["properties", "initialConditions", "anyOf"],
-)
+class _FluidDynamicsVolumeZone(VolumeZoneBase):
+    """FluidDynamicsVolumeZone type"""
+
+    model_type = pd.Field("FluidDynamics", alias="modelType", const=True)
+    reference_frame: Optional[
+        Union[
+            ReferenceFrame,
+            ReferenceFrameExpression,
+            ReferenceFrameDynamic,
+        ]
+    ] = pd.Field(alias="referenceFrame")
```

Comment:
are we able to achieve this by using `SchemaConfig.exclude...` or something similar?

Reply:
Issue with exclude is that we would need to delete the parent based on, let's say - the title, as members of an Union are treated as array elements - this would have to be handled via some other config field and the above just seemed like the most convenient way to limit schema-specific members.